untrusted/src/lib.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+// See https://download.01.org/intel-sgx/sgx-dcap/1.13/linux/docs/Intel_SGX_Enclave_Common_Loader_API_Reference.pdf
+//
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+
+use std::{ffi::CString, mem::MaybeUninit, os::raw::c_int};
+
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+#[derive(Default)]
+pub struct Enclave {
+    // The filename of the enclave
+    filename: CString,
+
+    // The enclave ID, assigned by the sgx interface
+    // Will be `None` when the enclave has not been created.
+    id: Option<sgx_enclave_id_t>,
+
+    // `true` if the enclave should be created in debug mode
+    debug: bool,
+}
+
+impl Enclave {
+    /// Returns an Enclave for the provided signed enclave.
+    ///
+    /// # Arguments
+    ///
+    /// * `filename` - The name of the enclave file.  This should be a signed
+    ///     enclave.
+    pub fn new(filename: &str) -> Enclave {
+        let filename = CString::new(filename).expect("Can't convert enclave filename to CString.");
+        Enclave {
+            filename,
+            ..Default::default()
+        }
+    }
```

Comment:
Few things:
1. We should be creating enclaves from bytes, not files, so we can bake them into the app via `include_bytes!()`. So this should take a byte slice (`&[u8]`) instead.
2. When doing file-based APIs, you should use `&Path` instead of `&str`, and `PathBuf` instead of `String`.
3. As noted below, this should be fallible and include the contents of `create()`

Reply:
Converted to use [sgx_create_enclave_from_buffer_ex](https://download.01.org/intel-sgx/sgx-linux/2.16/docs/Intel_SGX_Developer_Reference_Linux_2.16_Open_Source.pdf#%5B%7B%22num%22%3A232%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C94.5%2C339%2C0%5D)

Note: that the interface had to use a `&mut [u8]`, see https://github.com/mobilecoinfoundation/sgx/blob/3d0c0876bdcf512f5e8db894cd4a624a80d009a0/untrusted/src/lib.rs#L84